3p/integration.js

```diff
@@ -375,21 +387,22 @@ const defaultAllowedTypesInCustomFrame = [
 
 
 /**
- * Gets data encoded in iframe name attribute.
- * @return {!JsonObject}
+ * Initialize 3p frame.
+ * @param {!Window} win
  */
-function getData(iframeName) {
-  try {
-    // TODO(bradfrizzell@): Change the data structure of the attributes
-    //    to make it less terrible.
-    return parseJson(iframeName)['attributes'];
-  } catch (err) {
-    if (!getMode().test) {
-      dev().info(
-          'INTEGRATION', 'Could not parse context from:', iframeName);
-    }
-    return FALLBACK_CONTEXT_DATA;
-  }
+function init(win) {
+  const data = getData();
```

Comment:
yes, but there is no need to couple them in the same parsing. if you really want to save a JSON parsing (which is reasonable), we can introduce:
```
const config = parseNameConfig();
...
window.AMP_MODE = config._context.mode;
...
const context = config._context;
...
```

GTW, I do think we should rename `getData` funciton.  `data` has a different meaning there than `context.data`